ui/src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -263,7 +263,10 @@ const SearchBar = (props: {content: string; ctx: ContextApis; apps: models.Appli
                 </React.Fragment>
             )}
             onSelect={val => {
-                ctx.navigation.goto(`./${val}`);
+                const selectedApp = apps.find(app => AppUtils.appQualifiedName(app, useAuthSettingsCtx?.appsInAnyNamespaceEnabled) === val);
```

Comment:
```suggestion
                const qualifiedName = AppUtils.appQualifiedName(app, useAuthSettingsCtx?.appsInAnyNamespaceEnabled);
                const selectedApp = apps?.find(app => qualifiedName === val);
```

I'd suggest assigning it to a temporary variable to clarify what you're matching against.